applications/GeoMechanicsApplication/custom_utilities/README.md

```diff
@@ -28,9 +37,74 @@ File transport_equation_utilities.hpp includes
 -  CalculatePermeabilityMatrix function
 -  CalculateCompressibilityMatrix function
 
+## Stress strain utilities
+
+For convenience functions that compute invariants, equivalents and strain definitions.
+Given a stress tensor $\sigma$ or a strain tensor $\epsilon$. The eigenvalues of the stress tensor are $\sigma_1 \le \sigma_2 \le \sigma_3$
+
+$$\sigma = \begin{bmatrix} \sigma_{xx} & \sigma_{xy} & \sigma_{xz} \\
+                           \sigma_{xy} & \sigma_{yy} & \sigma_{yz} \\
+                           \sigma_{xz} & \sigma_{yz} & \sigma_{zz}  \end{bmatrix}$$
+
+$$\epsilon = \begin{bmatrix} \epsilon_{xx} & \epsilon_{xy} & \epsilon_{xz} \\
+                             \epsilon_{xy} & \epsilon_{yy} & \epsilon_{yz} \\
+                             \epsilon_{xz} & \epsilon_{yz} & \epsilon_{zz}  \end{bmatrix}$$
+
+### Trace
+
+The first tensor invariant:
+
+$$I_1 = trace(\sigma) = \Sigma_i \sigma_{i,i}$$
+
+### Mean stress
+
+$$p = \frac{1}{3} I_1 = \frac{1}{3} trace ( \sigma )$$
+
+### Von Mises stress
+
+With $J_2$ the second invariant of the tensor:
+
+$$\overline\sigma = \sqrt{3 J_2} = \sqrt{0.5((\sigma_{xx}-\sigma_{yy})^2 +
+                                 (\sigma_{yy}-\sigma_{zz})^2 +
+                                 (\sigma_{zz}-\sigma_{xx})^2 ) +
+                            3.0(\sigma_{xy}^2 + \sigma_{yz}^2 + \sigma_{xz}^2) }$$
+
+### Von Mises strain
+
+$$\overline{\epsilon} = \frac{2}{3} \sqrt{3 J_2}$$
+
+### Green Lagrange strain tensor
+
+With current configuration $x$ and reference configuration $X$, the deformation gradient $F = \frac{x}{X}$ and unit tensor I:
+
+$$\epsilon = 0.5 ( F^T \cdot F - I )$$
+
+### Hencky strain tensor
+
+$$\epsilon = 0.5 \ln ( F^T \cdot F )$$
+
+### Lode angle
+
+The negative sine definition for Lode angle is adapted here [Lode coordinates Wikipedia](https://en.wikipedia.org/wiki/Lode_coordinates):
+
+$$- \sin( 3 \bar{\theta}_s) = \frac{J_3}{2} (\frac{3}{J_2})^{\frac{3}{2}}$$
+
+which brings:
+
+$$\bar{\theta}_s = \frac{1}{3} asin( - \frac{27}{2} \frac{(\sigma_1 - p)(\sigma_2 - p)(\sigma_3 - p)}{q^3})$$
```

Comment:
Shouldn't we use `$\arcsin$` rather than `$asin$`, to change $asin$ to $\arcsin$?

Reply:
Done